stock_deferred_assign/models/stock_move.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import timedelta
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    reservation_date = fields.Date(
+        "Date to Reserve",
+        compute="_compute_reservation_date",
+        store=True,
```

Comment:
This compute will be triggered for all existing moves, so that can be killing when installing the module.